force-app/main/default/classes/DMLOperations.cls

```diff
@@ -149,6 +161,16 @@ public with sharing class DMLOperations {
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+            Account acc = new Account(Name = accountName);
+            insert acc;
+            for (String name : oppNames){
+                Opportunity opp = new Opportunity();
+                opp.Name = name;
+                opp.StageName = 'Qualification';
+                opp.CloseDate = Date.today().addMonths(3);
+                opp.AccountId = acc.Id;
+                upsert opp;
```

Comment:
Nice job! this will  upsert the opp but it's doing it one at a time, Try to create a list and add each opp to the list then upsert the list. This helps avoid governor limits.